[BUG] No indication in error of how to fix "DoNotOptimize<long long>' is deprecated: The const-ref version ..."

**Describe the bug**
I recently updated the version of this repository used by my bazel(isk) WORKSPACE and I started encountering the deprecation warning about the const-ref form of DoNotOptimize. While I appreciate the warning and want to do better, it's unclear to me what I need to do differently to make it go away.

**System**
Which OS, compiler, and compiler version are you using:
  - OS: OSX (uname -v: Darwin Kernel Version 19.6.0)
  - Compiler and version: llvm (probably. whatever bazelisk picks)

**To reproduce**
Steps to reproduce the behavior:
1. sync to commit d29044d5da4b32e449073a365ad6524fb9bdf884
2. bazelisk test :empty_test
3. See error: "problems/open/empty_test.cc:19:16: warning: 'DoNotOptimize<long long>' is deprecated: The const-ref version of this method can permit undesired compiler optimizations in benchmarks [-Wdeprecated-declarations]"

**Expected behavior**
I would like a description (or link to documentation) about what I need to change in that program to be getting ahead of the deprecation.

**Additional context**
BUILD
```
load("@rules_cc//cc:defs.bzl", "cc_test")
load("//euler/computer_science:data_file.bzl", "data_cc_file")

cc_test(
    name = "empty_test",
    timeout = "short",
    srcs = ["empty_test.cc"],
    tags = ["benchmark"],
    deps = [
        "//euler/computer_science:vlog",
        "@com_google_absl//absl/log",
        "@com_google_absl//absl/log:check",
        "@com_google_googletest//:gtest",
        "@com_monkeynova_gunit_main//:test_main",
    ],
)

```

empty_test.cc
```
#include "absl/log/check.h"
#include "absl/log/log.h"
#include "benchmark/benchmark.h"
#include "euler/computer_science/vlog.h"
#include "gtest/gtest.h"

int64_t RunProblem(int64_t range) {
  int64_t ret = -1;
  VLOG(1) << "RunProblem(" << range << ") = " << ret;
  return ret;
}

TEST(Euler, Given) { EXPECT_EQ(RunProblem(-1), -1); }

TEST(Euler, Problem) { EXPECT_EQ(RunProblem(-1), -1); }

void BM_ProblemXXX(benchmark::State& state) {
  for (auto _ : state) {
    benchmark::DoNotOptimize( RunProblem(state.range(0)));
  }
}

BENCHMARK(BM_ProblemXXX)->Arg(-1);
```

--- a/docs/user_guide.md
+++ b/docs/user_guide.md
@@ -1462,10 +1462,10 @@ known. For example:
   // while (...) DoNotOptimize(__result__);
 ```
 
-Since that is not the behavior the user intended, such problematic cases
-produce a compile-time diagnostic. Pass a non-const lvalue instead. For an
-expression result, store the result in a local variable before calling
-`DoNotOptimize`:
+Since that is not the behaviour the user intended,
+such problematic cases will result in a diagnostic
+at compile time. The correct approach, for an expression result,
+is to use an intermediate local variable:
 
 ```c++
   // Avoid: may call the deprecated const-reference overload.

--- a/docs/user_guide.md
+++ b/docs/user_guide.md
@@ -1462,12 +1462,12 @@ known. For example:
   // while (...) DoNotOptimize(__result__);
 ```
 
-Passing a temporary or `const` value may call the deprecated const-reference
-overload. Store the value in a local non-const variable and pass that lvalue to
-make the result observable:
+If `DoNotOptimize` warns about the deprecated const-reference overload, pass it
+a non-const lvalue instead. For an expression result, store the result in a
+local variable before calling `DoNotOptimize`:
 
 ```c++
-  // Avoid: calls the deprecated const-reference overload for the temporary.
+  // Avoid: may call the deprecated const-reference overload.
   while (...) DoNotOptimize(foo(0));
 
   // Prefer: materialize the result, then pass the local lvalue.

--- a/docs/user_guide.md
+++ b/docs/user_guide.md
@@ -1462,9 +1462,10 @@ known. For example:
   // while (...) DoNotOptimize(__result__);
 ```
 
-If `DoNotOptimize` warns about the deprecated const-reference overload, pass it
-a non-const lvalue instead. For an expression result, store the result in a
-local variable before calling `DoNotOptimize`:
+Since that is not the behavior the user intended, such problematic cases
+produce a compile-time diagnostic. Pass a non-const lvalue instead. For an
+expression result, store the result in a local variable before calling
+`DoNotOptimize`:
 
 ```c++
   // Avoid: may call the deprecated const-reference overload.

--- a/docs/user_guide.md
+++ b/docs/user_guide.md
@@ -1462,6 +1462,21 @@ known. For example:
   // while (...) DoNotOptimize(__result__);
 ```
 
+Passing a temporary or `const` value may call the deprecated const-reference
+overload. Store the value in a local non-const variable and pass that lvalue to
+make the result observable:
+
+```c++
+  // Avoid: calls the deprecated const-reference overload for the temporary.
+  while (...) DoNotOptimize(foo(0));
+
+  // Prefer: materialize the result, then pass the local lvalue.
+  while (...) {
+    auto result = foo(0);
+    DoNotOptimize(result);
+  }
+```
+
 The second tool for preventing optimizations is `ClobberMemory()`. In essence
 `ClobberMemory()` forces the compiler to perform all pending writes to global
 memory. Memory managed by block scope objects must be "escaped" using

--- a/include/benchmark/utils.h
+++ b/include/benchmark/utils.h
@@ -37,12 +37,15 @@ inline BENCHMARK_ALWAYS_INLINE void ClobberMemory() {
   std::atomic_signal_fence(std::memory_order_acq_rel);
 }
 
+#define BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG                     \
+  "DoNotOptimize(T const&) can permit undesired compiler optimizations. "    \
+  "Store the expression result in a local non-const variable and pass that " \
+  "lvalue instead."
+
 #ifndef BENCHMARK_HAS_NO_INLINE_ASSEMBLY
 #if !defined(__GNUC__) || defined(__llvm__) || defined(__INTEL_COMPILER)
 template <class Tp>
-BENCHMARK_DEPRECATED_MSG(
-    "The const-ref version of this method can permit "
-    "undesired compiler optimizations in benchmarks")
+BENCHMARK_DEPRECATED_MSG(BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG)
 inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
   asm volatile("" : : "r,m"(value) : "memory");
 }
@@ -66,9 +69,7 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp&& value) {
 }
 #elif (__GNUC__ >= 5)
 template <class Tp>
-BENCHMARK_DEPRECATED_MSG(
-    "The const-ref version of this method can permit "
-    "undesired compiler optimizations in benchmarks")
+BENCHMARK_DEPRECATED_MSG(BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG)
 inline BENCHMARK_ALWAYS_INLINE
     typename std::enable_if<std::is_trivially_copyable<Tp>::value &&
                             (sizeof(Tp) <= sizeof(Tp*))>::type
@@ -77,9 +78,7 @@ inline BENCHMARK_ALWAYS_INLINE
 }
 
 template <class Tp>
-BENCHMARK_DEPRECATED_MSG(
-    "The const-ref version of this method can permit "
-    "undesired compiler optimizations in benchmarks")
+BENCHMARK_DEPRECATED_MSG(BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG)
 inline BENCHMARK_ALWAYS_INLINE
     typename std::enable_if<!std::is_trivially_copyable<Tp>::value ||
                             (sizeof(Tp) > sizeof(Tp*))>::type
@@ -122,9 +121,7 @@ inline BENCHMARK_ALWAYS_INLINE
 
 #elif defined(_MSC_VER)
 template <class Tp>
-BENCHMARK_DEPRECATED_MSG(
-    "The const-ref version of this method can permit "
-    "undesired compiler optimizations in benchmarks")
+BENCHMARK_DEPRECATED_MSG(BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG)
 inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
   internal::UseCharPointer(&reinterpret_cast<char const volatile&>(value));
   ClobberMemory();
@@ -147,6 +144,8 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp&& value) {
   internal::UseCharPointer(&reinterpret_cast<char const volatile&>(value));
 }
 #endif
+
+#undef BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG
 
 }  // end namespace benchmark
 

--- a/include/benchmark/utils.h
+++ b/include/benchmark/utils.h
@@ -37,8 +37,8 @@ inline BENCHMARK_ALWAYS_INLINE void ClobberMemory() {
   std::atomic_signal_fence(std::memory_order_acq_rel);
 }
 
-#define BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG                     \
-  "DoNotOptimize(T const&) can permit undesired compiler optimizations. "    \
+#define BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG                       \
+  "DoNotOptimize(T const&) can permit undesired compiler optimizations. "      \
   "Pass a non-const lvalue instead; if the argument is an expression result, " \
   "store it in a local variable first."
 

--- a/include/benchmark/utils.h
+++ b/include/benchmark/utils.h
@@ -39,8 +39,8 @@ inline BENCHMARK_ALWAYS_INLINE void ClobberMemory() {
 
 #define BENCHMARK_DONOTOPTIMIZE_CONST_REF_DEPRECATED_MSG                     \
   "DoNotOptimize(T const&) can permit undesired compiler optimizations. "    \
-  "Store the expression result in a local non-const variable and pass that " \
-  "lvalue instead."
+  "Pass a non-const lvalue instead; if the argument is an expression result, " \
+  "store it in a local variable first."
 
 #ifndef BENCHMARK_HAS_NO_INLINE_ASSEMBLY
 #if !defined(__GNUC__) || defined(__llvm__) || defined(__INTEL_COMPILER)

